### ISSUE TITLE
Routing score drops on binary update (takes days to recover)

### ISSUE THREAD

**Issue by wiiinnie** [nymtech/nym#4233]
**Describe the issue**
After updating the mixnode binaries the routing score drops for various hours to days resulting in the mixnodes not getting into active set.

**Expected behaviour**
After a binary upgrade a node should recover their routing score within a short period of time

**Stack Traces**
Mixnode's logs look completely normal

**Steps to Reproduce**
- Upgrade node to latest version
- Restart process
- Monitor the routing score in explorer (official or Guru)

**Screenshots**
A node with nearly 50% saturation hasn't been in active set for 48h already:

![Screenshot 2023-12-08 at 10 08 21](https://github.com/nymtech/nym/assets/59703082/396ee06e-d71a-4917-bc80-b3d8b233e46d)
![Screenshot 2023-12-08 at 10 08 32](https://github.com/nymtech/nym/assets/59703082/c31bf7f9-3899-47db-8722-543da9c556e5)
![Screenshot 2023-12-08 at 10 08 40](https://github.com/nymtech/nym/assets/59703082/6c67f210-2447-452f-a8f2-313d5581eee3)

**Which area of Nym were you using?**
- Nymtech explorer and/or Guru
- Mixnode 1.1.34
- Ubunutu

**Comment by serinko** [comment]
I will clarify a few things:

1. Nym devs are constantly looking for improvement and working on the issues mentioned. We do read the issues and have been addressing them. And I hope that this issue can be resolved soon. At the same time note that.

2. Routing score is measured by a route of 3 hops. DPv2 brought a unexpected number of new nodes - many of them with poor setup (many restarts, tweaks etc) while at the same time many operators upgraded at once which they usually don't do. This resulted in a cascading effect of worsening routing score across the network. 

3. Many operators after a few hours of the slower up-time recovery (because of the dynamic explained in point no. 2) started to down-grade their nodes and when realizing that it didn't change the situation (hence it wasn't a binary problem per se), they upgraded again. This makes the problem worse, especially when a sentiment is shared that others shall down-grade as well. (If I am not mistaken this was also a case of @wiiinnie 's nodes shown above. 

4. DPv2 new nodes also result in a much larger competition as the active set is still 240, but the number of online nodes increased drastically. That is a good thing for the health of the network.

**Comment by wiiinnie** [comment]
> I will clarify a few things:
> 
> 1. Nym devs are constantly looking for improvement and working on the issues mentioned. We do read the issues and have been addressing them. And I hope that this issue can be resolved soon. At the same time note that.

Looking forward to it...

> 2. Routing score is measured by a route of 3 hops. DPv2 brought a unexpected number of new nodes - many of them with poor setup (many restarts, tweaks etc) while at the same time many operators upgraded at once which they usually don't do. This resulted in a cascading effect of worsening routing score across the network.

Thanks for the heads-up @serinko. However the question is if there is a way to mitigate this routing score issue. I do understand it was "bad timing" that many operators upgraded on the same day, is this simply "well, that's how it is, we can't change it"? Let me rephrase my questions: Isn't there anything we can change about how routing score is measured to make sure properly set up nodes recover more quickly? With weekly updated of the binaries we will have this issue each and every week...

> 3. Many operators after a few hours of the slower up-time recovery (because of the dynamic explained in point no. 2) started to down-grade their nodes and when realizing that it didn't change the situation (hence it wasn't a binary problem per se), they upgraded again. This makes the problem worse, especially when a sentiment is shared that others shall down-grade as well. (If I am not mistaken this was also a case of @wiiinnie 's nodes shown above.

I down graded one of three nodes, the other two stayed on 1.1.34 and haven't been restarted after the initial upgrade. So no, this is not the case. And btw the one that was reverted back to 1.1.33 has the highest routing score of the three even it has been online uninterruptedly only for approx 36h, as compared to the 1.1.34 nodes which have been online uninterruptedly for 48h...?

> 4. DPv2 new nodes also result in a much larger competition as the active set is still 240, but the number of online nodes increased drastically. That is a good thing for the health of the network.

In general yes. But it's not healthy for the network in the long run if the nodes that actually try to stay up to date loose their delegators who switch to nodes in the active set although they are several versions behind. Right?

**Comment by serinko** [comment]
> > I will clarify a few things:
> > 
> > 1. Nym devs are constantly looking for improvement and working on the issues mentioned. We do read the issues and have been addressing them. And I hope that this issue can be resolved soon. At the same time note that.
> 
> Looking forward to it...
> 
> > 2. Routing score is measured by a route of 3 hops. DPv2 brought a unexpected number of new nodes - many of them with poor setup (many restarts, tweaks etc) while at the same time many operators upgraded at once which they usually don't do. This resulted in a cascading effect of worsening routing score across the network.
> 
> Thanks for the heads-up @serinko. However the question is if there is a way to mitigate this routing score issue. I do understand it was "bad timing" that many operators upgraded on the same day, is this simply "well, that's how it is, we can't change it"? Let me rephrase my questions: Isn't there anything we can change about how routing score is measured to make sure properly set up nodes recover more quickly? With weekly updated of the binaries we will have this issue each and every week...
> 
This is exactly what we have to look into and have on the to-do list. I will keep you update it as it's an issue are addressing.

> > 3. Many operators after a few hours of the slower up-time recovery (because of the dynamic explained in point no. 2) started to down-grade their nodes and when realizing that it didn't change the situation (hence it wasn't a binary problem per se), they upgraded again. This makes the problem worse, especially when a sentiment is shared that others shall down-grade as well. (If I am not mistaken this was also a case of @wiiinnie 's nodes shown above.
> 
> I down graded one of three nodes, the other two stayed on 1.1.34 and haven't been restarted after the initial upgrade. So no, this is not the case. And btw the one that was reverted back to 1.1.33 has the highest routing score of the three even it has been online uninterruptedly only for approx 36h, as compared to the 1.1.34 nodes which have been online uninterruptedly for 48h...?
> 
I am aware that this has been an issue for quite a few, however others did not experience that. There are nodes who did the upgrade and were back the same day. So it wasn't this bad for everyone. This is a good thing as well as it leaves the question what are the different variables between different nodes.

> > 4. DPv2 new nodes also result in a much larger competition as the active set is still 240, but the number of online nodes increased drastically. That is a good thing for the health of the network.
> 
> In general yes. But it's not healthy for the network in the long run if the nodes that actually try to stay up to date loose their delegators who switch to nodes in the active set although they are several versions behind. Right?

On this one, I don't think we gonna face such a mess in the coming weeks time. There had been 304 new nodes setup in 10 days + others upgrading at the same time. The number of new nodes (and tbh many bad ones - try and error setup with ppl trying to harvest tokens) is unusual and it's only because of the program submissions.

**Comment by tommyv1987** [comment]
@all parties, interested, we're about to release new binaries, which shall fix the routing score issue. I'm going to close this ticket in respect to this. If any issues arise, we can re-open this ticket.
